src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1792,11 +1792,11 @@ public void clear(StyleInfo info) {
      * @return A reader for the style.
      */
     public BufferedReader readStyle( StyleInfo style ) throws IOException {
-        File styleFile = dataDir().findStyleSldFile(style);
-        if( styleFile == null ) {
-            throw new IOException( "No such file: " + style.getFilename() );
+        Resource styleResource = dataDir().style(style);
```

Comment:
Can you provide me some details here?  Any of these methods scare me as they often have odd/subtle consequences...

A bit surprised it was missed previously.